widget/embedded/src/hooks/useSyncUrlAndStore/useSyncUrlAndStore.helpers.ts

```diff
@@ -1,30 +1,22 @@
-import type { BlockchainMeta, Token } from 'rango-sdk';
+import type { Asset, BlockchainMeta, Token } from 'rango-sdk';
 
-export function searchParamToToken(
-  tokens: Token[],
+export function getTokenAsset(
   searchParam: string | null,
   chain: BlockchainMeta | null
-): Token | null {
+): Asset | undefined {
   if (!chain) {
```

Comment:
I don't get it why `searchParam` and `chain` should have value to this function, but we are accepting `null` values? Please remove `null` from inputs and handle them outside the function.